modules/twinkleblock.js

```diff
@@ -251,11 +251,13 @@ Twinkle.block.callback.change_block64 = function twinkleblockCallbackChangeBlock
 	var priorName = relevantUserName;
 	if ($block64.is(':checked')) {
 		relevantUserName = Morebits.ip.get64(mw.config.get('wgRelevantUserName'));
-		$form.find('[name=actiontype][value=template]').prop('disabled', true).prop('checked', false);
 	} else {
 		relevantUserName = mw.config.get('wgRelevantUserName');
-		$form.find('[name=actiontype][value=template]').prop('disabled', Morebits.ip.isRange(relevantUserName)).prop('checked', !Morebits.ip.isRange(relevantUserName));
 	}
+  // No templates for ranges, but if the original user is a single IP, offer the option
+  // (done separately in Twinkle.block.callback.issue_template)
+  var originalIsRange = Morebits.ip.isRange(mw.config.get('wgRelevantUserName'));
+  $form.find('[name=actiontype][value=template]').prop('disabled', originalIsRange).prop('checked', !originalIsRange);
```

Comment:
Am I correct that this should happen for both cases?

Reply:
Yeah — both to offer the template for the single IPv6 if the box is checked (as per this PR), and as normal if the box is unchecked.